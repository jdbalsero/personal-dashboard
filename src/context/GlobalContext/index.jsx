import React, { createContext, useState, useEffect } from 'react';

const GlobalContext = createContext();

function PageProvider(props) {
    const [activePage, setActivePage] = useState(() => {
        // Initialize activePage from localStorage or set a default value
        const savedActivePage = localStorage.getItem('activePage');
        return savedActivePage || 'Home';
    });

    const [activeBlur, setActiveBlur] = useState(() => {
        // Initialize activeBlur from localStorage or set a default value
        const savedActiveBlur = localStorage.getItem('activeBlur');
        return savedActiveBlur || '';
    });


    useEffect(() => {
        // Save state values to localStorage whenever they change
        localStorage.setItem('activePage', activePage);
        localStorage.setItem('activeBlur', activeBlur);
    }, [activePage, activeBlur]);

    return (
        <GlobalContext.Provider
            value={{
                activePage,
                setActivePage,
                activeBlur,
                setActiveBlur,
            }}
        >
            {props.children}
        </GlobalContext.Provider>
    );
}

export { GlobalContext, PageProvider };


