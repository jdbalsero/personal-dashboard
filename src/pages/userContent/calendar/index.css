.event-add-button{
    display: flex;
    width: 15%;
    height: 35px;
    justify-content: center;
    align-items: center;
    border-radius: 8px;
    border-color: transparent;
    font-weight: bold;
    color: #FFF;
    background: var(--main-blue-color);
    cursor: pointer;
    margin-bottom: 6px;
}