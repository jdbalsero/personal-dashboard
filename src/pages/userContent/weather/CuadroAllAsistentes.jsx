import React from 'react'
import imgDefaultUser from '../../../assets/images/programador.png';
import { IoIosStar } from 'react-icons/io';
import './AllAssistants.css'

function CuadroAllAsistentes(props) {

    const { asistente } = props;
    return (
        <div className='cuadroInfo sombra2' style={{background: asistente.bgColor}}>
            <div className='name-container'>
                <div className="assistant-name">
                    <span>{asistente.nombre}</span>
                </div>
            </div>
            <div className='img-container'>
                <img className='imgUser' src={imgDefaultUser} alt="usuario" />
            </div>
            <div className='rating-box'>
                {Array.from({ length: asistente.calification }).map((element, index) => 
                     <div key={index}>
                        <IoIosStar className='rating-star'/>
                     </div>
                )}
            </div>
            <div>
                <p><strong>Email: </strong> {asistente.correo}</p>
            </div>
        </div>
    )
}


export { CuadroAllAsistentes };