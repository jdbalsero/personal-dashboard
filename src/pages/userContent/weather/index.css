.weather-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  
  .weather-box {
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap:25px;
    background-color: #FFFFFF;
    border-radius: 10px;
    box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.1);
    padding: 20px;
  }

  .temperature {
    font-size: 50px;
    margin: 0;
  }
  
  .secondary-text {
    font-size: 20px;
    margin: 0;
  }

  .details-text {
    font-size: 18px;
    margin: 0;
  }

  .icon-text{
    display:flex;
    align-items: center;
    gap:2px;
    margin:0;
  }