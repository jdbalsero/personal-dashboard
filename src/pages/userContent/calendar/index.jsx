import React, { useState, useEffect, useContext } from 'react';
import { Popup } from '../../../components/Shared/Popup';
import Backdrop from '@mui/material/Backdrop';
import CircularProgress from '@mui/material/CircularProgress';
import { GlobalContext } from "../../../context/GlobalContext";
import { ContentHeader } from '../../../components/Shared/ContentHeader';
import { CalendarBox } from '../../../components/Shared/Calendar';
import { PopupCreateEvent } from './PopupCreateEvent';
import { Button } from "../../../components/Shared/Button";
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import './index.css';
const MySwal = withReactContent(Swal);


function CalendarClient() {
    const [events, setEvents] = useState([]);
    const [isPopupOpen, setIsPopupOpen] = useState(false);
    const { setActiveBlur } = useContext(GlobalContext);

    const fetchEvents = async () => {
        try {
            let dataEvents = JSON.parse(localStorage.getItem("Events")) === null ? [] : JSON.parse(localStorage.getItem("Events"));
            const eventsCalendar = await dataEvents.map((item, index) => {
                return {
                    id: index,
                    title: item.title,
                    start: new Date(item.start),
                    end: new Date(item.end),
                    description: item.description,
                }
            });
            setEvents(eventsCalendar);
        } catch (error) {
            console.error(error);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error retrieving the tasks, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    };

    useEffect(() => {
        fetchEvents();
    }, []);

    const createButtonClick = () => {
        setIsPopupOpen(true);
        setActiveBlur(true);
    }

    const handleAddEvent = (data) => {
        try{
            events.push(data);
            localStorage.setItem("Events", JSON.stringify(events));
            setActiveBlur(true);
            MySwal.fire({
                icon: 'success',
                title: 'Success',
                text: 'Event Saved.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
            closePopup();
            fetchEvents();
        }catch(err){
            console.error(err);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error adding the event, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    }

    const closePopup = () => {
        setIsPopupOpen(false);
        setActiveBlur(false);
    }

    return <React.Fragment>
        <div className="calendar">
            <div className="calendar-header">
                <ContentHeader text="Calendar" />
            </div>
            <div className="calendar-body">
                { events !== undefined ? 
                    <>
                        <Popup isOpen={isPopupOpen} onClose={closePopup} modalClass="popup-info-task">
                            <PopupCreateEvent
                                handleAddEvent = {handleAddEvent}
                                closePopup = {closePopup}
                            />
                        </Popup>
                        <Button
                            buttonClassName="event-add-button"
                            onClick={createButtonClick}
                            children="Add Event"
                        />
                        <CalendarBox events={events} />
                    </>
                    :
                    <Backdrop
                        sx={{ color: '#fff', zIndex: (theme) => theme.zIndex.drawer + 1 }}
                        open={true}
                    >
                        <CircularProgress style={{ height: "50px", width: "100px" }} color="inherit" />
                    </Backdrop>
                }
            </div>
        </div>
    </React.Fragment>
}

export { CalendarClient };