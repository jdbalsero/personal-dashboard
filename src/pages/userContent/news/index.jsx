import React, { useEffect, useState, useContext } from 'react';
import { GlobalContext } from '../../../context/GlobalContext';
import { ContentHeader } from '../../../components/Shared/ContentHeader'
import { TableNews } from './TableNews';
import TextField from '@mui/material/TextField';
import Autocomplete from '@mui/material/Autocomplete';
import './news.css';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
const MySwal = withReactContent(Swal);

function NewsPage() {
    const Categories = ['General', 'World', 'Business', 'Entertainment', 'Science', 'Sports', 'Health', 'Technology'];
    const { setActiveBlur } = useContext(GlobalContext);
    const [news, setNews] = useState([])
    const [value, setValue] = React.useState(Categories[0]);
    const [inputValue, setInputValue] = React.useState('');

    const showNews = async (category=null) => {
        const API_BASE_URL = category === null ? 
        `https://gnews.io/api/v4/top-headlines?category=general&lang=en&apikey=${process.env.REACT_APP_GNEWSAPI_KEY}`
        : `https://gnews.io/api/v4/top-headlines?lang=en&category=${category}&apikey=${process.env.REACT_APP_GNEWSAPI_KEY}`;
        try {
            const response = await fetch(API_BASE_URL).then(response => {
              return response;
            });
            if (!response.ok) {
                throw new Error('Error en la solicitud');
            }
            let data = await response.json();
            setNews(data.articles);
        } catch (error) {
            console.error(error);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was an error searching news, please check your filter selection and try again.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
      }

    useEffect(() => {
        showNews();
    }, [])

    return <div className="data-news-container">
        <ContentHeader text="News" />
        <div className="search-section">
            <Autocomplete
                disablePortal
                value={value}
                onChange={(event, newValue) => {
                    setValue(newValue);
                    showNews(String(newValue).toLowerCase());
                }}
                inputValue={inputValue}
                onInputChange={(event, newInputValue) => {
                    setInputValue(newInputValue);
                }}
                id="controllable-states-demo"
                className='input-search'
                options={Categories}
                sx={{ width: 300 }}
                renderInput={(params) => <TextField {...params} label="News Category" />}
            />
        </div>
        <TableNews data={news} fetchData={showNews} />
    </div>
}

export { NewsPage }