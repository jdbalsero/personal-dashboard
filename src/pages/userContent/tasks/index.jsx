import React, { useContext, useState, useEffect } from "react";
import { GlobalContext } from '../../../context/GlobalContext';
import { Button } from "../../../components/Shared/Button";
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import Paper from '@mui/material/Paper';
import { styled } from '@mui/material/styles';
import IconButton from '@mui/material/IconButton';
import DeleteIcon from '@mui/icons-material/Delete';
import FormControlLabel from '@mui/material/FormControlLabel';
import OutlinedInput from '@mui/material/OutlinedInput';
import Checkbox from '@mui/material/Checkbox';
import { ContentHeader } from '../../../components/Shared/ContentHeader'
import './tasks_style.css';
const MySwal = withReactContent(Swal);

const Item = styled(Paper)(({ theme }) => ({
    backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
    ...theme.typography.body2,
    padding: theme.spacing(4),
    textAlign: 'center',
    color: theme.palette.text.secondary,
  }));
  

function TasksPage() {
    const [todolist_tasks, setToDoListTasks] = useState([]);
    const { setActiveBlur } = useContext(GlobalContext)
    const [inputValue, setInputValue] = useState('');

    const handleChange = (event) => {
        setInputValue(event.target.value);
    };

    const handleDeleteClick = (todo_id) => {
        setActiveBlur(true);
        MySwal.fire({
            icon:'warning',
            title: `Are you Sure?`,
            text: 'Do you want to delete this to do task ?',
            showDenyButton: true,
            showCancelButton: false,
            confirmButtonText: 'Yes, I want to delete it.',
            denyButtonText: `No, I want to keep it`,
            backdrop: true,
            customClass: {
                popup: 'popup-sweet',
                title: 'title-sweet',
                htmlContainer: 'text-sweet',
                confirmButton: 'confirm-button-sweet',
                denyButton: 'deny-button-sweet',
            }
        }).then((result) => {
            if (result.isConfirmed) {
                todolist_tasks.splice(todo_id,1);
                localStorage.setItem("Tasks", JSON.stringify(todolist_tasks));
                setActiveBlur(true);
                MySwal.fire({
                    icon: 'success',
                    title: 'Success',
                    text: 'To Do Task Deleted.',
                    showConfirmButton: false,
                    timer: 1700,
                    backdrop: true,
                    customClass: {
                        popup: 'popup-sweet',
                        title: 'title-sweet',
                        htmlContainer: 'text-sweet',
                        confirmButton: 'confirm-button-sweet',
                        denyButton: 'deny-button-sweet',
                    }
                }).finally(() => {setActiveBlur(false)});
                fetchData();
            }
        }).finally(() => {setActiveBlur(false)});
    };

    const handleAddToDoTask = async() => {
        try{
            todolist_tasks.push({description: inputValue, status:"Active"});
            localStorage.setItem("Tasks", JSON.stringify(todolist_tasks));
            setActiveBlur(true);
            MySwal.fire({
                icon: 'success',
                title: 'Success',
                text: 'To Do List Task Saved.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
            setInputValue('');
            fetchData();
        }catch(err){
            console.error(err);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error adding the new task, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    }

    const handleUpdateStatus = async (id, value) => {
        try{
            todolist_tasks[id].status = value ? "Completed":"Active";
            localStorage.setItem("Tasks", JSON.stringify(todolist_tasks));
            fetchData();
        }catch(err){
            console.error(err);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error updating the task, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    }

    const fetchData = async () => {
        try {
            let newRows = JSON.parse(localStorage.getItem("Tasks")) === null ? [] : JSON.parse(localStorage.getItem("Tasks"));
            setToDoListTasks(newRows);
        } catch (error) {
            console.error(error);
            setActiveBlur(true);
            MySwal.fire({
                icon: 'error',
                title: 'Oops ...',
                text: 'There was a unexpected error retrieving the tasks, please contact support.',
                showConfirmButton: false,
                timer: 1700,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
        }
    };

    const handleKeyDown = (e) => {
        if (e.key === 'Enter') {
          handleAddToDoTask();
        }
      }

    useEffect(() => {
        fetchData();
    }, []);

    return (
        <React.Fragment>
            <div className="todo-general-container">
                <ContentHeader text="Tasks" />
                {todolist_tasks.length > 0 ? 
                    <div className="list-task">
                        <List sx={{ width: '100%' }}>
                            {todolist_tasks.map((option, index) => {
                                const labelId = `item-list-label-${option}`;
                                return (
                                <div key={index}>
                                    <ListItem className="task-item" disablePadding secondaryAction={
                                        <IconButton edge="end" aria-label="delete" onClick={() => {handleDeleteClick(index);}}>
                                            <DeleteIcon className="icon-color"/>
                                        </IconButton>
                                    }>
                                        <ListItemIcon className="task-checkbox-container">
                                            <FormControlLabel
                                                control={
                                                    <Checkbox
                                                        checked={option.status === "Active" ? false : true}
                                                        style ={{
                                                            color: "#5873FE",
                                                        }}
                                                        onChange={e => handleUpdateStatus(index, e.target.checked)}
                                                    />
                                                }
                                            />
                                        </ListItemIcon>
                                        <ListItemText id={labelId} primary={option.description} className={option.status === "Completed" ? 'linethrough' : ''}  secondary={`Status: ${option.status}`}/>
                                    </ListItem>
                                </div>
                                );
                            })}
                        </List>
                    </div>
                : <Box sx={{ width: '100%' }} style={{ display: 'flex', justifyContent: 'center', margin:'17px' }}>
                        <Typography
                            sx={{ display: 'inline' }}
                            component="span"
                            variant="body2"
                            color="text.primary"
                        >
                            You don't have any task in the list
                        </Typography>
                    </Box>
                }
                <div className="todo-add-button-container">
                    <OutlinedInput className="inputTodo" type="text" placeholder="Type the task you need" value={inputValue} onChange={handleChange} onKeyDown={handleKeyDown} />
                    <Button
                        buttonClassName="todo-add-button"
                        onClick={() => handleAddToDoTask() }
                        children="Add Task"
                    />
                </div>
            </div>
        </React.Fragment>
    );
}

export { TasksPage };