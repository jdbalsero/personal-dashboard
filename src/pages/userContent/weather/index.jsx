import React, { useState, useEffect, useContext} from 'react';
import { GlobalContext } from '../../../context/GlobalContext';
import { ContentHeader } from '../../../components/Shared/ContentHeader';
import { SearchComponent } from '../../../components/Shared/SearchComponent';
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import './index.css';
const MySwal = withReactContent(Swal);

function WeatherPage() {
  const { setActiveBlur } = useContext(GlobalContext);
  const [loading, setLoading] = useState(true);
  const [response, setResponse] = useState({});
  const [search, setSearch] = useState("");

  const searcher = (e) => {
    setSearch(e.target.value)   
  }

  const searchSpecificData = async (e) => {
    const API_BASE_URL = `https://api.openweathermap.org/data/2.5/weather?q=${search}&units=metric&appid=${process.env.REACT_APP_OPENWEATHERAPI_KEY}`;
    try {
        const response = await fetch(API_BASE_URL).then(response => {
          return response;
        });
        if (!response.ok) {
            throw new Error('Error en la solicitud');
        }
        let data = await response.json();
        setResponse({
          temperature: data.main.temp,
          condition: data.weather[0].main,
          description: data.weather[0].description,
          location: data.name+', '+data.sys.country,
          icon: data.weather[0].icon,
          feel: data.main.feels_like,
          humidity: data.main.humidity,
          pressure: data.main.pressure,
          temp_max: data.main.temp_max,
          temp_min: data.main.temp_min,
          wind_speed: data.wind.speed,
          wind_degree_direction: data.wind.deg,
          wind_gust: data.wind.gust,
          visibility: data.visibility
        });
    } catch (error) {
        console.error(error);
        setActiveBlur(true);
        MySwal.fire({
            icon: 'error',
            title: 'Oops ...',
            text: 'There was an error searching the location, please check the location name and try again.',
            showConfirmButton: false,
            timer: 1700,
            backdrop: true,
            customClass: {
                popup: 'popup-sweet',
                title: 'title-sweet',
                htmlContainer: 'text-sweet',
                confirmButton: 'confirm-button-sweet',
                denyButton: 'deny-button-sweet',
            }
        }).finally(() => {setActiveBlur(false)});
    }
  }

  const fetchDataInitial = async (lat, lon) => {
    const API_BASE_URL = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=metric&appid=${process.env.REACT_APP_OPENWEATHERAPI_KEY}`;
    try {
        const response = await fetch(API_BASE_URL).then(response => {
          return response;
        });
        if (!response.ok) {
            throw new Error('Error en la solicitud');
        }
        let data = await response.json();
        setResponse({
          temperature: data.main.temp,
          condition: data.weather[0].main,
          description: data.weather[0].description,
          location: data.name+', '+data.sys.country,
          icon: data.weather[0].icon,
          feel: data.main.feels_like,
          humidity: data.main.humidity,
          pressure: data.main.pressure,
          temp_max: data.main.temp_max,
          temp_min: data.main.temp_min,
          wind_speed: data.wind.speed,
          wind_degree_direction: data.wind.deg,
          wind_gust: data.wind.gust,
          visibility: data.visibility
        });
    } catch (error) {
        console.error(error);
        setActiveBlur(true);
        MySwal.fire({
            icon: 'error',
            title: 'Oops ...',
            text: 'There was an error searching the location, please check the location name and try again.',
            showConfirmButton: false,
            timer: 1700,
            backdrop: true,
            customClass: {
                popup: 'popup-sweet',
                title: 'title-sweet',
                htmlContainer: 'text-sweet',
                confirmButton: 'confirm-button-sweet',
                denyButton: 'deny-button-sweet',
            }
        }).finally(() => {setActiveBlur(false)});
    }
  }

  var options = {
    enableHighAccuracy: true,
    timeout: 5000,
    maximumAge: 0,
  };

  function success(pos) {
    var crd = pos.coords;
    fetchDataInitial(crd.latitude, crd.longitude);
    setLoading(false);
  }

  function errors(err) {
    console.warn(`ERROR(${err.code}): ${err.message}`);
  }

  useEffect(() => {
    if (navigator.geolocation) {
      navigator.permissions
        .query({ name: "geolocation" })
        .then(function (result) {
          if (result.state === "granted") {
            navigator.geolocation.getCurrentPosition(success, errors, options);
          } else if (result.state === "prompt") {
            navigator.geolocation.getCurrentPosition(success, errors, options);
          } else if (result.state === "denied") {
            setActiveBlur(true);
            MySwal.fire({
                icon: 'warning',
                title: 'Oops ...',
                text: 'Geolocation not allowed, this component may not function.',
                showConfirmButton: true,
                backdrop: true,
                customClass: {
                    popup: 'popup-sweet',
                    title: 'title-sweet',
                    htmlContainer: 'text-sweet',
                    confirmButton: 'confirm-button-sweet',
                    denyButton: 'deny-button-sweet',
                }
            }).finally(() => {setActiveBlur(false)});
          }
        });
    } else {
      setActiveBlur(true);
      MySwal.fire({
          icon: 'warning',
          title: 'Oops ...',
          text: 'Geolocation is not supported by this browser.',
          showConfirmButton: false,
          timer: 1700,
          backdrop: true,
          customClass: {
              popup: 'popup-sweet',
              title: 'title-sweet',
              htmlContainer: 'text-sweet',
              confirmButton: 'confirm-button-sweet',
              denyButton: 'deny-button-sweet',
          }
      }).finally(() => {setActiveBlur(false)});
    }
  }, []);

    return (
      <React.Fragment>
        <div className="weather-header" style={{display:'flex',flexDirection:'row',alignItems:'center',padding:'20px',gap:'10px',flex:'none',order:'0', alignSelf:'stretch',flexGrow:'0'}}>
          <ContentHeader text="Real-Time Weather" />
        </div>
        { loading ? (
          <><p>Loading... (You must accept the location permission in your browser)</p></>  
          ) : (
            <>
              <div className="weather-container">
                <div className="weather-box">
                  <div className="principal-column">
                    <p className="secondary-text">{response.location}</p>
                    <h2 className="temperature">{response.temperature ? response.temperature+'°' : '0°'}</h2>
                    <p className="icon-text">
                      <img className="img-icon" src={`http://openweathermap.org/img/w/${response.icon ? response.icon : '03d'}.png`} alt="Weather icon"/>
                      <span className="secondary-text">{response.condition}</span>
                    </p>
                  </div>
                  <div className="details-column-a">
                    <p className="details-text"><b>Temp feel:</b> {response.feel ? response.feel+'°' : '0°'}</p>
                    <p className="details-text"><b>Temp Max:</b> {response.temp_max ? response.temp_max+'°' : '0°'}</p>
                    <p className="details-text"><b>Temp Min:</b> {response.temp_min ? response.temp_min+'°' : '0°'}</p>
                    <p className="details-text"><b>Visibility:</b> {response.visibility} m</p>
                  </div>
                  <div className="details-column-b">
                    <p className="details-text"><b>Humidity:</b> {response.humidity}%</p>
                    <p className="details-text"><b>Pressure:</b> {response.pressure} hPa</p>
                    <p className="details-text"><b>Wind Speed:</b> {response.wind_speed} m/s</p>
                    <p className="details-text"><b>Wind Direction (Degree):</b> {response.wind_degree_direction? response.wind_degree_direction+'°' : '0°'}</p>
                  </div>
                </div>
              </div>
              <div className="search-container">
                <div className="text-search-container">
                  <span>Search specific locations:</span>
                </div>
                <SearchComponent search={search} searcher={searcher} placeholder={"Location name"} onclick={searchSpecificData}/>
              </div>
            </>
          )}
      </React.Fragment>
    );
}

export { WeatherPage };
