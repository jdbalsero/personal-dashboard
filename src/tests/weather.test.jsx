import { render, screen } from '@testing-library/react';
import { WeatherPage} from '../pages/userContent/weather';
import { PageProvider } from '../context/GlobalContext';
import React from 'react';


describe('<WeatherPage /> Tests', () => {

    test('Match Snapshot', () => {
        const { container } = render(
        <PageProvider>
            <WeatherPage />
        </PageProvider>);
        expect( container ).toMatchSnapshot();
    });

    test('Show Title of Page', () => {
        render(
            <PageProvider>
                <WeatherPage />
            </PageProvider>
        );
        expect( screen.getByText("Real-Time Weather") ).toBeTruthy();
    });

    test('Show Permission Message until user access', () => {

        render(
            <PageProvider>
                <WeatherPage />
            </PageProvider>
        );

        expect( screen.getByText("Loading... (You must accept the location permission in your browser)") ).toBeTruthy();
    });
});