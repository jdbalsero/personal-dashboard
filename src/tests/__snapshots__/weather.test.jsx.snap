// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherPage /> Tests Match Snapshot 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="weather-header"
    style="display: flex; flex-direction: row; align-items: center; padding: 20px; gap: 10px; flex: 0 0 auto; order: 0; align-self: stretch; flex-grow: 0;"
  >
    <h1
      class="ContentHeader"
      style="color: rgb(88, 115, 254); font-weight: normal;"
    >
      Real-Time Weather
    </h1>
  </div>
  <p>
    Loading... (You must accept the location permission in your browser)
  </p>
</div>
`;
